### ISSUE TITLE
`rest` route matcher

### ISSUE THREAD

**Issue by rnett** [hfhbd/routing-compose#206]
I want to get the rest of a path from the given route, not matching if there aren't any more segments.  For example:
```kotlin
route("foo") {
   rest { path -> println(path) }
}
```

Would not match `/foo` (that would fall down to `noMatch`), would match `/foo/bar` and print `bar`, and would match `/foo/bar/baz` and print `bar/baz` (I'm not stuck on the slash placement).

`string` is similar currently, but only matches a single path segment.  `noMatch` matches these routes, but doesn't provide the sub-path (and has its own distinct semantics).

**Comment by rnett** [comment]
TIL about `remainingPath`, does does do what I wanted, thanks.

**Comment by hfhbd** [comment]
Yeah, it is a special attribute for the `NoMatch` class of the `noMatch` handler :)
